### PR TITLE
Add special case for null start end values for numerical study view filters

### PR DESCRIPTION
**Pull request by gblaih**
Fix https://github.com/cBioPortal/cbioportal/issues/11633

Add special case where if null start and end values are passed in a numerical filter on study view, don't apply a filter. Prevents sql error where empty tuple () occurs.

### CODE REVIEW

**Comment by haynescd** [comment]
I don't think I understand the issue? Can you add an integration test here that validates this edge case has been fixed

**Review by haynescd — CHANGES_REQUESTED** [review]
Need tests

File: src/main/resources/mappers/clickhouse/studyview/ClickhouseStudyViewFilterMapper.xml [hunk 1]
```diff
@@ -280,6 +280,12 @@
                     AND 
                     ${attribute_value} ILIKE #{dataFilterValue.value}
                 </if>
+                <!--
+                    Special case: In case null start, end, and value values go through, don't apply filter.
+                -->
+                <if test="dataFilterValue.start == null and dataFilterValue.end == null">
```
Comment (onursumer) [inline src/main/resources/mappers/clickhouse/studyview/ClickhouseStudyViewFilterMapper.xml:286]: The remaining `if test` conditions seem to cover this case so we shouldn't be applying any filter. I'm surprised we still somehow apply filters when everything is null.

File: src/main/java/org/cbioportal/domain/studyview/StudyViewFilterContext.java [hunk 2]
```diff
@@ -52,7 +52,16 @@ public String[] filteredSampleIdentifiers() {
   }
 
   public boolean isCategoricalClinicalDataFilter(ClinicalDataFilter clinicalDataFilter) {
-    var filterValue = clinicalDataFilter.getValues().getFirst();
+    var firstValue = clinicalDataFilter.getValues().getFirst();
+    var lastValue = clinicalDataFilter.getValues().getLast();
+    // there is a case where a categorical filter will have a set of null filter values first
```
Comment (onursumer) [inline src/main/java/org/cbioportal/domain/studyview/StudyViewFilterContext.java:57]: Ideally we should filter out invalid filters before passing them to this function, probably in the controller or service layer, so that we don't have to add more logic here.

File: src/main/resources/mappers/clickhouse/studyview/ClickhouseStudyViewFilterMapper.xml (hunk 1 above)
Comment (gblaih) [reply src/main/resources/mappers/clickhouse/studyview/ClickhouseStudyViewFilterMapper.xml:286]: In the case where everything is null, none of the if conditions will pass so no filter should be applied. However, this `applyNumericalDataFilter` sql block is constructed like `((dataFilterValue1), (dataFilterValue2), ...)` and if everything is null in the original code, we will get an empty tuple which causes the sql errors we were encountering. So I added this condition so that it would act as if no filter is passed while also preventing the empty tuple sql error.
